pkg/cmd/roachprod/grafana/annotations.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/util/httputil"
+	"github.com/cockroachdb/errors"
+	"github.com/go-openapi/strfmt"
+	grafclient "github.com/grafana/grafana-openapi-client-go/client"
+	"github.com/grafana/grafana-openapi-client-go/models"
+	"google.golang.org/api/idtoken"
+)
+
+const ServiceAccountJson = "GRAFANA_SERVICE_ACCOUNT_JSON"
+
+// newGrafanaClient is a helper function that creates an HTTP client to
+// create grafana api calls with.  If secure is true, it tries to get a
+// GCS identity token by using the service account specified by the env
+// variable ServiceAccountJson. This identity token is passed in
+// every request to authenticate with grafana.
+func newGrafanaClient(
+	ctx context.Context, host string, secure bool,
+) (*grafclient.GrafanaHTTPAPI, error) {
+	const grafanaAudience = "1063333028845-p47csl1ukrgnpnnjc7lrtrto6uqs9t37.apps.googleusercontent.com"
```

Comment:
While this appears to be sticky (i.e., unlikely to change frequently), maybe this should also be injected via the env?

Reply:
Done.